lib/removeUnusedElements.js

```diff
@@ -158,6 +166,62 @@ Remove.bufferView = function(gltf, bufferViewId) {
     }
 };
 
+Remove.mesh = function(gltf, meshId) {
+    const meshes = gltf.meshes;
+    meshes.splice(meshId, 1);
+
+    ForEach.node(gltf, function(n) {
+        if(defined(n.mesh)) {
+            if(n.mesh > meshId) {
+                --n.mesh;
+            } else if(n.mesh === meshId) {
+                /* Remove reference to deleted mesh */
+                delete n.mesh;
+            }
+        }
+    });
+};
+
+Remove.node = function(gltf, nodeId) {
+    const nodes = gltf.nodes;
+
+    /* Remove this node from parent */
+    const parentId = findParentNode(gltf, nodeId);
+    if(parentId >= 0) {
+        nodes[parentId].children = nodes[parentId].children.filter(x => x !== nodeId);
+    }
+
+    nodes.splice(nodeId, 1);
+
+    /* Shift all node references */
+    nodes.forEach(other => {
+        if(defined(other.children)) {
+            other.children = other.children.map(x => x > nodeId ? x - 1 : x);
+        }
+    });
+    ForEach.scene(gltf, (s, i) => {
+        s.nodes = s.nodes
+            .filter(x => x !== nodeId) /* Remove */
+            .map(x => x > nodeId ? x - 1 : x); /* Shift indices */
```

Comment:
Could this same filter/map code be used above to combine the remove-from-parent and shift-indices steps? Maybe something like:

```js
    const nodes = gltf.nodes;
    nodes.splice(nodeId, 1);

    ForEach.node(gltf, (n) => {
        n.children = n.children
            .filter(x => x !== nodeId) /* Remove */
            .map(x => x > nodeId ? x - 1 : x); /* Shift indices */
    });
```